pkg/migration/file.go

```diff
@@ -96,6 +96,20 @@ func (m *MigrationFile) ExecBatch(ctx context.Context, conn *pgx.Conn) error {
 			if len(pgErr.Detail) > 0 {
 				msg = append(msg, pgErr.Detail)
 			}
+			// Provide helpful hint for extension type errors (SQLSTATE 42704: undefined_object)
+			if pgErr.Code == "42704" && strings.Contains(pgErr.Message, "type") && strings.Contains(pgErr.Message, "does not exist") {
+				// Extract type name from error message (e.g., 'type "ltree" does not exist')
+				typeName := extractTypeName(pgErr.Message)
+				msg = append(msg, "")
+				msg = append(msg, "Hint: This type may be defined in a schema that's not in your search_path.")
+				msg = append(msg, "      Use schema-qualified type references to avoid this error:")
+				if typeName != "" {
+					msg = append(msg, fmt.Sprintf("        CREATE TABLE example (col extensions.%s);", typeName))
+				} else {
+					msg = append(msg, "        CREATE TABLE example (col extensions.<type_name>);")
+				}
```

Comment:
```suggestion
			if typeName := extractTypeName(pgErr.Message); len(typeName) > 0 && pgErr.Code == "42704" {
				msg = append(msg, "")
				msg = append(msg, "Hint: This type may be defined in a schema that's not in your search_path.")
				msg = append(msg, "      Use schema-qualified type references to avoid this error:")
				msg = append(msg, fmt.Sprintf("        CREATE TABLE example (col extensions.%s);", typeName))
```